http.go

```diff
@@ -41,41 +42,34 @@ func startHTTPHandler(state *State) {
 			c.AbortWithStatus(http.StatusForbidden)
 			return
 		}
-		c.Next()
-	}, gin.LoggerWithFormatter(func(param gin.LogFormatterParams) string {
-		var statusColor, methodColor, resetColor string
-		if param.IsOutputColor() {
-			statusColor = param.StatusCodeColor()
-			methodColor = param.MethodColor()
-			resetColor = param.ResetColor()
+
+		// Return if the hostname is an IP
+		hostname := strings.Split(c.Request.Host, ":")[0]
+		hostnameIsIp := net.ParseIP(hostname)
+		if hostnameIsIp != nil {
+			c.AbortWithStatus(http.StatusForbidden)
+			return
 		}
 
+		c.Next()
+	}, gin.LoggerWithFormatter(func(param gin.LogFormatterParams) string {
 		if param.Latency > time.Minute {
 			// Truncate in a golang < 1.8 safe way
 			param.Latency = param.Latency - param.Latency%time.Second
 		}
 
-		logLine := fmt.Sprintf("%v | %s |%s %3d %s| %13v | %15s |%s %-7s %s %s\n%s",
-			param.TimeStamp.Format("2006/01/02 - 15:04:05"),
-			param.Request.Host,
-			statusColor, param.StatusCode, resetColor,
-			param.Latency,
-			param.ClientIP,
-			methodColor, param.Method, resetColor,
-			param.Path,
-			param.ErrorMessage,
-		)
-
+		// Client log
 		if *logToClient {
 			hostname := strings.Split(param.Request.Host, ":")[0]
 			loc, ok := state.HTTPListeners.Load(hostname)
 			if ok {
 				proxyHolder := loc.(*ProxyHolder)
-				sendMessage(proxyHolder.SSHConn, strings.TrimSpace(logLine), true)
+				sendMessage(proxyHolder.SSHConn, strings.TrimSpace(formatLog(clientLogFormatParts, &param)), true)
 			}
 		}
 
-		return logLine
+		// Server log line
+		return formatLog(serverLogFormatParts, &param) + "\n"
```

Comment:
Move the newline to the `formatLog` method. That's why TrimSpace is used when sending the message to the channel.